core/src/main/scala/org/apache/spark/sql/execution/ExistingPlans.scala

```diff
@@ -262,41 +262,47 @@ case class ExecutePlan(child: SparkPlan, preAction: () => Unit = () => ())
     val session = sqlContext.sparkSession.asInstanceOf[SnappySession]
-    val sc = session.sparkContext
-    val key = session.currentKey
-    val oldExecutionId = sc.getLocalProperty(SQLExecution.EXECUTION_ID_KEY)
-    val (result, shuffleIds) = if (oldExecutionId eq null) {
-      val (queryStringShortForm, queryStr, queryExecStr, planInfo) = if (key eq null) {
-        val callSite = sqlContext.sparkContext.getCallSite()
-        (callSite.shortForm, callSite.longForm, treeString(verbose = true),
+    try {
+      val sc = session.sparkContext
+      val key = session.currentKey
+      val oldExecutionId = sc.getLocalProperty(SQLExecution.EXECUTION_ID_KEY)
+      val (result, shuffleIds) = if (oldExecutionId eq null) {
+        val (queryStringShortForm, queryStr, queryExecStr, planInfo) = if (key eq null) {
+          val callSite = sqlContext.sparkContext.getCallSite()
+          (callSite.shortForm, callSite.longForm, treeString(verbose = true),
             PartitionedPhysicalScan.getSparkPlanInfo(this))
-      } else {
-        val paramLiterals = key.currentLiterals
-        val paramsId = key.currentParamsId
-        (key.sqlText, key.sqlText, SnappySession.replaceParamLiterals(
-          treeString(verbose = true), paramLiterals, paramsId), PartitionedPhysicalScan
+        } else {
+          val paramLiterals = key.currentLiterals
+          val paramsId = key.currentParamsId
+          (key.sqlText, key.sqlText, SnappySession.replaceParamLiterals(
+            treeString(verbose = true), paramLiterals, paramsId), PartitionedPhysicalScan
             .getSparkPlanInfo(this, paramLiterals, paramsId))
-      }
-      CachedDataFrame.withNewExecutionId(session, queryStringShortForm,
-        queryStr, queryExecStr, planInfo) {
+        }
+        CachedDataFrame.withNewExecutionId(session, queryStringShortForm,
+          queryStr, queryExecStr, planInfo) {
+          preAction()
+          val rdd = child.execute()
+          val shuffleIds = SnappySession.findShuffleDependencies(rdd)
+          (collectRDD(sc, rdd), shuffleIds)
+        }._1
+      } else {
         preAction()
         val rdd = child.execute()
         val shuffleIds = SnappySession.findShuffleDependencies(rdd)
         (collectRDD(sc, rdd), shuffleIds)
-      }._1
-    } else {
-      preAction()
-      val rdd = child.execute()
-      val shuffleIds = SnappySession.findShuffleDependencies(rdd)
-      (collectRDD(sc, rdd), shuffleIds)
-    }
-    if (shuffleIds.nonEmpty) {
-      sc.cleaner match {
-        case Some(c) => shuffleIds.foreach(c.doCleanupShuffle(_, blocking = false))
-        case None =>
       }
+      if (shuffleIds.nonEmpty) {
+        sc.cleaner match {
+          case Some(c) => shuffleIds.foreach(c.doCleanupShuffle(_, blocking = false))
+          case None =>
+        }
+      }
+      result
+    }
+    finally {
+      logDebug(" Unlocking the table in execute of ExecutePlan ")
```

Comment:
Could add the table name to log here. Perhaps available in session else maybe child.treeString(verbose = false)